### DISCUSSION TITLE
What if we make multi class intent classification?

### DISCUSSION THREAD

**Discussion by OnlinePage** [iparsw/custom-intent#9] [Ideas]
So far now the ChatBot class performs very well with good accuracy and speed. I was thinking of making it more versatile by able to return Multiclass labels or say intents for the messages.
for example :- sentence: Tell me my account balance? Also please update my passbook?
it has two clear intents: 1> balance_enquiry and 2>passbook_update

now generally the model is capable of classify single intents, so makit in a way like above will make it more versatile.

Also is there any way that it can be achieve in the exisiting ChatBot class ?

**Comment by iparsw** [comment]
Yes theoretically it is possible the last layer is a softmax layer and it will return the probability of every intent and if it’s two intents that fit, the first and second most probable result will have close probability but it won’t work so well in practice and won’t get a good enough accuracy with this simple models, but that’s a great idea I don’t think i can work on it in near future as i said exams and our version of sat, but i will try to create a model that can do that without getting too complex and heavy.
